### ISSUE TITLE
Add accessibility tests to CI process

### ISSUE THREAD

**Issue by interactivellama** [salesforce/design-system-react#2107]
#2098 does not include a11y tests. It uses JSDOM and takes 15+ minutes to run—resulting in 26 min of CI time. :-( 

And currently there are only about 1/4 of components in the accessibility suite, so that will only grow in time. 

aXe tests should be run in Puppeteer for performance. https://github.com/nickcolley/jest-axe/issues/10

Image snapshots is currently using Puppeteer
https://www.npmjs.com/package/@storybook/addon-storyshots-puppeteer#specifying-options-to-screenshot-puppeteer-api

So it will probably be a merger of these APIs.

Other resources:
* https://marmelab.com/blog/2018/07/18/accessibility-performance-testing-puppeteer.html
* https://www.npmjs.com/package/axe-puppeteer
